server/auth.go

```diff
@@ -402,24 +402,26 @@ func (s *Server) processClientOrLeafAuthentication(c *client, opts *Options) boo
 
 	// Check if we have trustedKeys defined in the server. If so we require a user jwt.
 	if s.trustedKeys != nil {
-		if c.opts.JWT == "" {
+		if c.opts.JWT == "" && (c.opts.Nkey == "" || s.opts.SystemAccount == "") {
```

Comment:
s.opts.SystemAccount will be empty but by default we have system accounts now, so best to check directly against the server.

Reply:
Also, for the future, we forgot to mention to you that you should not access `s.opts.<any option>` directly since this can be racy with config reload. You get options from s.getOpts() and work on those. We already do get the options on this function.